### PR TITLE
Allow passing --session-user in the CLI

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/6567.

@findepi @alexjo2144 have you planned to test it somehow?

### CODE REVIEW

**Comment by findepi** [comment]
Could this be testable with `TestTrinoCli`?

File: client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java [hunk 1]
```diff
@@ -143,6 +143,9 @@
     @Option(names = "--session", paramLabel = "<session>", description = "Session property (property can be used multiple times; format is key=value; use 'SHOW SESSION' to see available properties)")
     public final List<ClientSessionProperty> sessionProperties = new ArrayList<>();
 
+    @Option(names = "--session-user", paramLabel = "<user>", description = "Impersonating session username")
```
Comment (wendigo) [inline client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:146]: I didn't come up with a better description. Suggestions?

**Comment by wendigo** [comment]
Ac @findepi

File: client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java (hunk 1 above)
Comment (findepi) [reply client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:146]: "Username to impersonate"

File: testing/trino-product-tests/src/main/java/io/trino/tests/cli/TestTrinoCli.java [hunk 2]
```diff
@@ -182,6 +182,15 @@ public void shouldExecuteEmptyListOfStatements()
         trino.waitForWithTimeoutAndKill();
     }
 
+    @Test(groups = CLI, timeOut = TIMEOUT)
+    public void shouldPassSessionUser()
+            throws Exception
+    {
+        launchTrinoCliWithServerArgument("--execute", "SELECT current_user;", "--session-user", "other-user");
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/cli/TestTrinoCli.java:189]: would read better is --session-user was before --execute

**Review by findepi — APPROVED** [review]

**Comment by wendigo** [comment]
@findepi I've pushed updated PR

**Pull request merged**